src/transmitter/index.ts

```diff
@@ -308,7 +306,12 @@ export async function retryRequest(
     let statusCode: number | undefined = undefined;
 
     try {
-      response = await needle(verb, url, payload, options);
+      if (payload) {
+        response = await needle(verb, url, payload, options);
+      } else {
+        response = await needle(verb, url, options);
```

Comment:
That's a very scary function signature! How does it know we're passing "options" and not "a payload with no options"? 😵

Reply:
yes you are right, I will change

Reply:
not sure about get requests though 🤔